### ISSUE TITLE
String interpolation with nested quotes breaks Razor blocks.

### ISSUE THREAD

**Issue by NTaylorMullen** [aspnet/Razor#401]
Today if you are to write the following:

```HTML
@{
    $"abc{"123\""}.def";
}
```
You end up having a mismatch in curly braces. This is due to Razor not understanding the `{` syntax in string interpolation and therefore parsing the strings incorrectly.

**Comment by dougbu** [comment]
@NTaylorMullen is this bug a bit larger? In particular does Razor understand `\"` in a quoted string and does the following get parsed correctly?
``` c#
@{
  var variable = "abc\"}.def";
}
```

**Comment by NTaylorMullen** [comment]
@dougbu Yup, Razor understands that. `\"` is understood when it thinks it's inside of a string :smile:, that's why string interpolation sends it for a loop.

**Comment by NTaylorMullen** [comment]
Starting work on this to enable full support for: https://github.com/aspnet/Razor/issues/338

**Comment by NTaylorMullen** [comment]
After investigating this issue further turns out the requirements to complete this are pretty high (need to understand C# close to entirety at the Tokenizer level). Will revisit this later.

**Comment by Eilon** [comment]
Moving to backlog because we feel these cases are not common in MVC views, and have trivial workarounds (e.g. use `string.Format()` instead of string interpolation).
